homeassistant/components/climate/spider.py

```diff
@@ -0,0 +1,147 @@
+        self._operation = STATE_IDLE
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        supports = SUPPORT_TARGET_TEMPERATURE
+
+        if self._master:
+            supports = supports | SUPPORT_OPERATION_MODE
+
+        return supports
+
+    @property
+    def name(self):
+        """Return the name of the thermostat, if any."""
+        return self._name
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._target_temperature
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        return self._min_temp
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        return self._max_temp
+
+    @property
+    def current_operation(self: ClimateDevice) -> str:
+        """Return current operation ie. heat, cool, idle."""
+        return self._operation
+
+    @property
+    def operation_list(self):
+        """Return the list of available operation modes."""
+        return OPERATION_LIST
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        if temperature is None:
+            return
+
+        self.client.set_temperature(self._thermostat, temperature)
+
+    def set_operation_mode(self, operation_mode):
+        """Set new target operation mode."""
+        self.client.set_operation_mode(self._thermostat,
+                                       HA_STATE_TO_SPIDER.get(operation_mode))
+
+    def update(self):
+        """Get the latest data."""
+        try:
+            # Only let the master thermostat refresh
+            # and let the others use the cache
+            thermostats = self.client.get_thermostats(
+                force_refresh=self._master)
+            for thermostat in thermostats:
+                if thermostat['id'] == self._id:
+                    self._thermostat = thermostat
+
+        except StopIteration:
+            _LOGGER.error("No data from the Itho Daalderop API")
+            return
+
+        for prop in self._thermostat['properties']:
+            if prop['id'] == 'AmbientTemperature':
+                self._current_temperature = float(prop['status'])
+            if prop['id'] == 'SetpointTemperature':
+                self._target_temperature = float(prop['status'])
+                self._min_temp = float(prop['min'])
+                self._max_temp = float(prop['max'])
+            if prop['id'] == 'OperationMode':
+                self._master = True
+                self._operation = SPIDER_STATE_TO_HA[prop['status']]
```

Comment:
Now we don't catch idle state.

Reply:
Correct. In their web interface there is no such thing as idle. Either it's heating (even when the set temperature is lower then the actual temperature) or cooling (even when your boiler can't cool, the button is present to set it to cool). I added this as a note to the docs.